velox/common/base/SimdUtil-inl.h

```diff
@@ -1068,6 +1068,86 @@ struct HalfBatchImpl<
   using Type = Batch64<T>;
 };
 
+template <typename T, typename U, typename A>
+struct ReinterpretBatch {
+#if XSIMD_WITH_SSE2
+  static xsimd::batch<T, A> apply(xsimd::batch<U, A> data, const xsimd::sse2&) {
+    return xsimd::batch<T, A>(data);
+  }
+#endif
+
+#if XSIMD_WITH_AVX
+  static xsimd::batch<T, A> apply(xsimd::batch<U, A> data, const xsimd::avx&) {
+    return xsimd::batch<T, A>(data);
+  }
+#endif
+};
+
+template <typename T, typename A>
+struct ReinterpretBatch<T, T, A> {
+  static xsimd::batch<T, A> apply(xsimd::batch<T, A> data, const A&) {
+    return data;
+  }
+};
+
+#if XSIMD_WITH_NEON || XSIMD_WITH_NEON64
+
+#ifdef VELOX_SIMD_DEFINE_REINTERPRET_BATCH_INT
+#error "Macro name collision: VELOX_SIMD_DEFINE_REINTERPRET_BATCH_INT"
+#endif
+
+// clang-format off
+#define VELOX_SIMD_DEFINE_REINTERPRET_BATCH_INT(width)          \
```

Comment:
This failed @Yuhta. Needs some more fix 👎. Thank you.

```c
/home/ubuntu/Yu-Velox/./velox/common/base/SimdUtil-inl.h:1100:54: error: '#' is not followed by a macro parameter
 1100 | #define VELOX_SIMD_DEFINE_REINTERPRET_BATCH_INT(width)          \
      |                                                      ^
```

Reply:
Trying to be too smart here.  Can you check it again?